tests/simulationTest.m

```diff
@@ -0,0 +1,108 @@
+classdef simulationTest < matlab.unittest.TestCase
+    %Test for simulation
+    properties 
+        Config;
+        Simulation;
+    end
+    methods (TestClassSetup)
+        function createObjects(testCase)
+            testCase.Config = MonsterConfig();
+
+            testCase.Simulation = Monster(testCase.Config);
+        end
+    end
+
+    methods (TestMethodTeardown)
+        function resetObjects(testCase)
+            %Test Teardown
+
+        end
+    end
+
+    methods (Test)
+        function testConstructor(testCase)
+            testCase.verifyTrue(isa(testCase.Simulation, 'Monster'));
+        end
+
+        function testSetupRound(testCase)
+            %Setup simulation for round 0
+            iRound = 0;
+            testCase.Simulation.setupRound(iRound);
+            testCase.verifyTrue(testCase.Config.Runtime.currentRound == iRound);
+            testCase.verifyTrue(testCase.Config.Runtime.currentTime == iRound*10e-3);
+            testCase.verifyTrue(testCase.Config.Runtime.remainingTime == (testCase.Config.Runtime.totalRounds - testCase.Config.Runtime.currentRound)*10e-3);
+            testCase.verifyTrue(testCase.Config.Runtime.remainingRounds == testCase.Config.Runtime.totalRounds - testCase.Config.Runtime.currentRound - 1);
+
+            %Test for channel setup as well???
```

Comment:
Great idea. The thing lacking from the channel testing is integration tests with higher layer APIs such as the simulation loop.